realm/realm-library/src/main/java/io/realm/sync/permissions/Permission.java

```diff
@@ -0,0 +1,562 @@
+        /**
+         * Defines if this role is allowed to modify the schema of this resource.
+         *
+         * <ol>
+         *     <li>
+         *         <b>Realm:</b>
+         *         If {@code true} the role is allowed to create classes in the Realm.
+         *     </li>
+         *     <li>
+         *         <b>Class:</b>
+         *         If {@code true}, the role is allowed to add properties to the specified class.
+         *     </li>
+         *     <li>
+         *         <b>Object:</b>
+         *         Not applicable.
+         *     </li>
+         * </ol>
+         *
+         * @param canModifySchema {@code true} if the role is allowed to modify the schema of this resource.
+         */
+        public Builder canModifySchema(boolean canModifySchema) {
+            this.canModifySchema = canModifySchema;
+            return this;
+        }
+
+        /**
+         * Creates the unmanaged {@link Permission} object.
+         */
+        public Permission build() {
+            return new Permission(
+                    role,
+                    canRead,
+                    canUpdate,
+                    canDelete,
+                    canSetPermissions,
+                    canQuery,
+                    canCreate,
+                    canModifySchema
+            );
+        }
+    }
+
+    private Role role;
+    private boolean canRead;
+    private boolean canUpdate;
+    private boolean canDelete;
+    private boolean canSetPermissions;
+    private boolean canQuery;
+    private boolean canCreate;
+    private boolean canModifySchema;
+
+    public Permission() {
+        // Required by Realm
+    }
+
+    /**
+     * Creates a set of privileges for the given role.
+     */
+    public Permission(Role role) {
+        this.role = role;
+    }
+
+    /**
+     * Creates a set of privileges for the given role.
+     */
+    private Permission(Role role, boolean canRead, boolean canUpdate, boolean canDelete, boolean canSetPermissions, boolean canQuery, boolean canCreate, boolean canModifySchema) {
+        this.role = role;
+        this.canRead = canRead;
+        this.canUpdate = canUpdate;
+        this.canDelete = canDelete;
+        this.canSetPermissions = canSetPermissions;
+        this.canQuery = canQuery;
+        this.canCreate = canCreate;
+        this.canModifySchema = canModifySchema;
+    }
+
+    /**
+     * Returns the role these privileges apply to.
+     *
+     * @return the role these privileges apply to.
+     */
+    public Role getRole() {
+        return role;
+    }
+
+    /**
+     * Returns {@code true} if the role is allowed to read the resource, {@code false} if not.
+     */
+    public boolean canRead() {
+        return canRead;
+    }
+
+    /**
+     * Defines if this role can read from given resource or not.
+     *
+     * <ol>
+     *     <li>
+     *         <b>Realm:</b>
+     *         The role is allowed to read all objects from the Realm. If {@code false}, the
```

Comment:
Not "all objects", but any object at all. If `canRead` is false, nothing is visible. If it is true, everything in the Realm may still happen to be inaccessible to that user.

`canRead` doesn't determine whether you can read everything, but determines whether you can read anything at all.